conformance_pack/iam.sp

```diff
@@ -2158,3 +2166,75 @@ query "iam_policy_custom_no_permissive_role_assumption" {
         not is_aws_managed;
   EOQ
 }
+
+query "iam_inline_policy_no_administrative_privileges" {
+  sql = <<-EOQ
+    with iam_resource_types as (
+      select
+        arn,
+        inline_policies_std,
+        name,
+        account_id,
+        region,
+        _ctx,
+        'iam_user' as type
+      from
+        aws_iam_user
+      union
+      select
+        arn,
+        inline_policies_std,
+        name,
+        account_id,
+        region,
+        _ctx,
+        'iam_role' as type
+      from
+        aws_iam_role
+      union
+      select
+        arn,
+        inline_policies_std,
+        name,
+        account_id,
+        region,
+        _ctx,
+        'iam_group' as type
+      from
+        aws_iam_group
+    ),
+    bad_policies as (
+      select
+        arn,
+        count(*) as statements_num
+      from
+        iam_resource_types,
+        jsonb_array_elements(inline_policies_std) as policy_std,
+        jsonb_array_elements(policy_std -> 'PolicyDocument' -> 'Statement') as s,
+        jsonb_array_elements_text(s -> 'Resource') as resource,
+        jsonb_array_elements_text(s -> 'Action') as action
+      where
+        s ->> 'Effect' = 'Allow'
+        and resource = '*'
+        and (
+          (action = '*'
+          or action = '*:*'
+          )
+        )
+      group by
+        arn
+    )
+    select
+      p.arn as resource,
+      case
+        when bad.arn is null then 'ok'
+        else 'alarm'
+      end status,
+      p.name || ' contains ' || coalesce(bad.statements_num,0)  ||
+        ' statements that allow action "*" on resource "*".' as reason
+      ${replace(local.common_dimensions_qualifier_global_sql, "__QUALIFIER__", "p.")}
```

Comment:
Does the resource support tags? If yes, I think we should add the tag dimensions as well.

Reply:
For aws_iam_role and aws_iam_user they support tagging but aws_iam_group does not support tagging and in this query, we are doing a union of the three tables, that is the reason we cannot use tags here.